libcxx/include/__type_traits/is_equality_comparable.h

```diff
@@ -10,9 +10,11 @@
 #define _LIBCPP___TYPE_TRAITS_IS_EQUALITY_COMPARABLE_H
 
 #include <__config>
+#include <__type_traits/enable_if.h>
```

Comment:
Can we update at least `libcxx/test/std/algorithms/alg.nonmodifying/alg.equal/equal.pass.cpp` to add a test where we exercise this new code path? Seems useful to ensure that it works.